src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -213,15 +213,31 @@ bool clang_cpp_convertert::get_var(const clang::VarDecl &vd, exprt &new_expr)
   return clang_c_convertert::get_var(vd, new_expr);
 }
 
-bool clang_cpp_convertert::get_function(
+bool clang_cpp_convertert::get_method(
   const clang::FunctionDecl &fd,
```

Comment:
We can force this to be a `CXXMethodDecl`. It would make sense given the new function name.

Reply:
Hi @mikhailramalho I'd suggest we have a second thought on splitting `get_function` into `get_function` and `get_method`, for the following reasons:

1. Since CXXMethodDecl is also a FunctionDecl, the current `get_function` serves as a top-level unified conversion point for standalone funcitons (i.e. non-methods) not associateed with a class and methods associated with a class. 

2. After splitting, we are calling `get_method` in `get_function`. I personally find it very confusing - what if we are converting a standalone function that isn't a class method? 

From a debugging point of view, I find the existing code structure is pretty intuitive - we got a overriding `clang_cpp_convert::get_function` doing some additional checks before calling the clang_c_convert::get_fuction  . So can we keep the `get_function`? 

For the missing conversion of copy assignment operator, I would just move the if-condition from `clang_c_frontend::get_function` to `clang_cpp_convert::get_function` and add the additional filtering clause for copy assignment operator.